### PR TITLE
Adapt to minor env changes from code review

### PR DESCRIPTION
**Pull request by graydon**
Adapting SDK to https://github.com/stellar/rs-soroban-env/pull/1077 -- needs to have its refs updated when that merges.

### CODE REVIEW

File: soroban-sdk/src/env.rs [hunk 1]
```diff
@@ -1283,6 +1283,27 @@ impl Env {
...
+    // paths kept out of the host). We delegate this to our env_impl and then,
+    // since our own Error type is Infallible, immediately throw it into either
+    // the env_impl's Error escalation path (if testing), or just plain panic.
+    #[cfg(not(target_family = "wasm"))]
+    fn error_from_error_val(&self, e: crate::Error) -> Self::Error {
+        let host_err = self.env_impl.error_from_error_val(e);
+        #[cfg(any(test, feature = "testutils"))]
+        self.env_impl.escalate_error_to_panic(host_err);
+        #[cfg(not(any(test, feature = "testutils")))]
+        panic!("{:?}", host_err);
```
Comment (leighmcculloch) [inline soroban-sdk/src/env.rs:1297]: Contracts and the SDK are never executed without tests in non-wasm mode. What use case are these two lines serving?

**Review by leighmcculloch — APPROVED** [review]

File: soroban-sdk/src/env.rs (hunk 1 above)
Comment (graydon) [reply soroban-sdk/src/env.rs:1297]: Generating an `Infallible` when not-testutils (escalate_error_to_panic is gated on testutils)

Comment (graydon) [reply soroban-sdk/src/env.rs:1297]: (i.e. just to make it compile)

Comment (graydon) [reply soroban-sdk/src/env.rs:1297]: (and like .. on the off chance some weirdo runs it in that configuration. nobody will, but there needs to be a code path there)

**Pull request merged**